webpack/containers/FormsEditContainer.js

```diff
@@ -12,14 +12,23 @@ import { responseSetProps } from '../prop-types/response_set_props';
 import _ from 'lodash';
 
 class FormsEditContainer extends Component {
-  componentWillMount() {
+  constructor(props) {
+    super(props);
+    let selectedFormSaver = this.props.saveForm;
     if (this.props.params.formId) {
       this.props.fetchForm(this.props.params.formId);
-    }    else {
+      if (this.props.params.action === 'edit') {
+        selectedFormSaver = this.props.saveDraftForm;
+      }
+    } else {
       this.props.newForm();
       this.props.params.formId = 0;
       this.props.params.action = 'new';
     }
+    this.state = {selectedFormSaver};
+  }
+
```

Comment:
Does this container need something like:
```
   componentDidUpdate(prevProps) {
     if(prevProps.params.formId != this.props.params.formId){
       this.props.fetchForm(this.props.params.formId);
     }
   }
```
Currently there is a bug if you are on a form /edit or /revise page and you try to use the nav bar to navigate to the /forms/new page from the top nav bar where the component is not re-rendering / the confirmation modal is not popping up